plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java

```diff
@@ -795,8 +782,27 @@ protected String getKubernetesClusterNodeNamePrefix() {
         return prefix;
     }
 
+    protected String getEtcdNodeNameForCluster() {
+        String prefix = kubernetesCluster.getName();
+        if (!NetUtils.verifyDomainNameLabel(prefix, true)) {
+            prefix = prefix.replaceAll("[^a-zA-Z0-9-]", "");
+            if (prefix.isEmpty()) {
+                prefix = kubernetesCluster.getUuid();
+            }
+        }
+        prefix = prefix + "-etcd" ;
+        if (prefix.length() > 40) {
+            prefix = prefix.substring(0, 40);
+        }
+        return prefix;
+    }
+
     protected KubernetesClusterVO updateKubernetesClusterEntry(final Long cores, final Long memory, final Long size,
-               final Long serviceOfferingId, final Boolean autoscaleEnabled, final Long minSize, final Long maxSize) {
+                                                               final Long serviceOfferingId, final Boolean autoscaleEnabled,
+                                                               final Long minSize, final Long maxSize,
+                                                               final KubernetesClusterNodeType nodeType,
+                                                               final boolean updateNodeOffering,
+                                                               final boolean updateClusterOffering) {
         return Transaction.execute((TransactionCallback<KubernetesClusterVO>) status -> {
             KubernetesClusterVO updatedCluster = kubernetesClusterDao.createForUpdate(kubernetesCluster.getId());
```

Comment:
The `updateKubernetesClusterEntry` method is calling `kubernetesClusterDao.createForUpdate` to update the k8s cluster VO. Apparently, this DAO's method returns the VO with its attributes set to `null`. 

The VO is then updated, persisted and returned. Eventually, this returned VO will be assigned to the `com.cloud.kubernetes.cluster.actionworkers.KubernetesClusterActionWorker#kubernetesCluster` attribute.

Previously, this flow wasn't causing any issues, since it was only executed once per cluster scaling. However, currently, it's executed for each cluster node type. Therefore, in the loop's second iteration, `com.cloud.kubernetes.cluster.actionworkers.KubernetesClusterActionWorker#kubernetesCluster` will contain some `null` attributes and will break the scaling flow.

One possible workaround for this scenario would be to, in the `updateKubernetesClusterEntry` method, return the `kubernetesClusterDao.findById(kubernetesCluster.getId())` after persisting the modifications performed on `updatedCluster`.